.github/workflows/open-version-bump-pr.yaml

```diff
@@ -84,41 +84,93 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           FILE_TO_COMMIT: package.json
-          VERSION_BRANCH: 'version-bump/${{ steps.tick-version.outputs.VERSION_TAG }}'
+          VERSION_BRANCH: "version-bump"
         run: |
-          # start with creating new branch 'version-bump/VERSION_TAG', also update the reference to the origin if it doesn't exist
+          # start with creating new branch 'version-bump', also update the reference to the origin if it doesn't exist
           git checkout -b $VERSION_BRANCH
           if ! gh api -X GET /repos/:owner/:repo/git/ref/heads/$VERSION_BRANCH >/dev/null 2>&1; then
-            echo "creating new branch"
+            echo "creating new branch $VERSION_BRANCH"
             gh api -X POST /repos/:owner/:repo/git/refs -f ref="refs/heads/main" -f sha="$GITHUB_SHA" -f "ref=refs/heads/$VERSION_BRANCH"
-
-            # move the branch pointer one commit backwards so that we can manually commit changes done by 'npm version ...' command
+          fi
+          
+          # get the content of package.json from $VERSION_BRANCH to get the current bump version
+          PACKAGE_JSON_CONTENT=$(gh api --method GET /repos/:owner/:repo/contents/$FILE_TO_COMMIT \
+              -f "ref=refs/heads/$VERSION_BRANCH" \
+              | jq -r '.content' \
+              | base64 -d)
+          
+          BUMP_VERSION=$(echo "$PACKAGE_JSON_CONTENT" | jq -r '.version')
+          NEW_VERSION="${{ steps.tick-version.outputs.VERSION_TAG }}"
+          
+          echo "BUMP_VERSION $BUMP_VERSION"
+          echo "NEW_VERSION $NEW_VERSION"
+          
+          #We need to compare and update the package version in version bump PR only if the new version is higher then current bump version. 
+          #For example if PR labeled as 'minor' is merged, then only 'major' labeled PR will overwrite the version change, not the 'patch' or 'minor' labeled PRs. 
+          
+          # Split versions into major, minor, and patch components
+          IFS='.' read -r -a bump <<< "$BUMP_VERSION"
+          IFS='.' read -r -a new <<< "$NEW_VERSION"
+          
+          HIGHEST_VERSION="equal"
+          
+          # Compare major version number
+          if (( bump[0] > new[0] )); then
+            HIGHEST_VERSION="bump"
+          elif (( bump[0] < new[0] )); then
+            HIGHEST_VERSION="new"
+          else
+            # Compare minor version number
+            if (( bump[1] > new[1] )); then
+              HIGHEST_VERSION="bump"
+            elif (( bump[1] < new[1] )); then
+              HIGHEST_VERSION="new"
+            else
+              # Compare patch version number
+              if (( bump[2] > new[2] )); then
+                HIGHEST_VERSION="bump"
+              elif (( bump[2] < new[2] )); then
+                HIGHEST_VERSION="new"
+              fi
+            fi
+          fi
```

Comment:
If I'm understanding the logic correctly, couldn't we consolidate this down to one conditional? Something like `bump[0] < new[0] || bump[1] < new[1] || bump[2] < new[2]`

Reply:
We could maybe just use that directly below where we check `HIGHEST_VERSION`

Reply:
I think this will not work for different levels. For example if the bump version is 1.1.5 and the new version is 1.0.9 it will assume that the new version is higher because of `|| bump[2] < new[2]` . Surely we can have all the conditions one-lined, but i left this for readability.
On the other hand i don't think we will have versions that will differ like this.

Reply:
Oh ya you're right! This is good then